File: app/lib/models/asset.dart
class Asset {
  final String assetId;
  final String name;
  final String type;
  final String location;
  final String? description;
  final String qrCodePayload;
  final String createdAt;

  const Asset({
    required this.assetId,
    required this.name,
    required this.type,
    required this.location,
    this.description,
    required this.qrCodePayload,
    required this.createdAt,
  });

  factory Asset.fromJson(Map<String, dynamic> j) => Asset(
        assetId: j['assetId'] as String,
        name: j['name'] as String,
        type: j['type'] as String,
        location: j['location'] as String,
        description: j['description'] as String?,
        qrCodePayload: j['qrCodePayload'] as String,
        createdAt: j['createdAt'] as String,
      );
}

File: src/MaintenanceSystem.Domain/Entities/Asset.cs
namespace MaintenanceSystem.Domain.Entities;

public class Asset
{
    public Guid AssetId { get; private set; }
    public string Name { get; private set; } = string.Empty;
    public string Type { get; private set; } = string.Empty;
    public string Location { get; private set; } = string.Empty;
    public string? Description { get; private set; }
    public string QrCodePayload { get; private set; } = string.Empty;
    public DateTime CreatedAt { get; private set; }

    private Asset() { }

    public static Asset Create(string name, string type, string location, string? description = null)
    {
        var id = Guid.NewGuid();
        return new Asset
        {
            AssetId = id,
            Name = name,
            Type = type,
            Location = location,
            Description = description,
            QrCodePayload = $"asset:{id}",
            CreatedAt = DateTime.UtcNow
        };
    }

    public void Update(string name, string type, string location, string? description)
    {
        Name = name;
        Type = type;
        Location = location;
        Description = description;
    }
}
